server/sonar-server/src/main/java/org/sonar/ce/settings/ProjectConfigurationFactory.java

```diff
@@ -36,11 +40,22 @@ public ProjectConfigurationFactory(Settings globalSettings, DbClient dbClient) {
     this.dbClient = dbClient;
   }
 
-  public Configuration newProjectConfiguration(String projectKey) {
+  public Configuration newProjectConfiguration(String projectKey, Optional<Branch> branch) {
+    String effectiveKey = generateProjectKey(projectKey, branch);
     Settings projectSettings = new ProjectSettings(globalSettings);
     dbClient.propertiesDao()
-      .selectProjectProperties(projectKey)
+      .selectProjectProperties(effectiveKey)
       .forEach(property -> projectSettings.setProperty(property.getKey(), property.getValue()));
     return new ConfigurationBridge(projectSettings);
   }
+
+  private static String generateProjectKey(String projectKey, Optional<Branch> branch) {
+    if (branch.isPresent()) {
+      Branch b = branch.get();
+      if (!b.isLegacyFeature() && !b.isMain() && b.getName().isPresent()) {
+        return generateBranchKey(projectKey, b.getName().get());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)